controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -288,6 +288,25 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
 				}
+			case "CephFilesystemSubVolumeGroup":
+				found := &rookCephv1.CephFilesystemSubVolumeGroup{ObjectMeta: metav1.ObjectMeta{
+					Name:      resource.Name,
+					Namespace: r.storageClassClaim.Namespace,
+				}}
+				err := controllerutil.SetOwnerReference(r.storageClassClaim, found, r.Client.Scheme())
+				if err != nil {
+					return reconcile.Result{}, err
+				}
```

Comment:
Please add the owner reference inside the `CreateOrUpdate` call. Otherwise the owner ref array will get overwritten by the `get` call inside  `CreateOrUpdate`